module/Calendar.js

```diff
@@ -0,0 +1,311 @@
+			}
+			const validatedEvent = this._eventValidator(event);
+			const validatedDate = this._dateValidator(event.date, event.time);
+			const timerId = setTimeout(
+				callback,
+				new Date(validatedDate).getTime() - Date.now()
+			);
+			const eventObj = {
+				id: this._getId(),
+				...validatedEvent,
+				timerId,
+				callback,
+			};
+			this._events.push(eventObj);
+			return 'Added successfully!';
+		} catch (error) {
+			console.log(error.message);
+		}
+	}
+
+	/**
+	 * The method deletes the event
+	 * @method deleteEvent
+	 * @param {string} id
+	 * Id of an existing event
+	 * @example
+	 * deleteEvent('f9ca-1baa-c970-35b4')
+	 */
+	deleteEvent(id) {
+		try {
+			const events = this._events;
+			const index = events.findIndex((el) => el.id === id);
+			if (index !== -1) {
+				clearTimeout(events[index].timerId);
+				events.splice(index, 1);
+			} else {
+				throw new Error('Event not found');
+			}
+			return 'Deleted successfully!';
+		} catch (error) {
+			console.log(error.message);
+		}
+	}
+
+	/**
+	 * The method deletes the event
+	 * @method updateEvent
+	 * @param {string} id
+	 * Id of an existing event
+	 * @param {object} nextEvent
+	 * Object of the new event
+	 * @param {string} [event.title
+	 * Title of the new event
+	 * @param {string} event.time
+	 * Time of the new event in the format "12:00:00"
+	 * @param {string} event.date
+	 * Date of the new event in the format "2022-04-06"
+	 * @param {function} [nextCallback]
+	 * Callback function that will be called when the specified date in the event occurs
+	 * @example
+	 * updateEvent('f9ca-1baa-c970-35b4', { title: 'birthday', date: '2022-04-06', time: '18:24:00'}, () => { console.log('Happy Birthday') })
+	 */
+	updateEvent(id, nextEvent, nextCallback) {
+		try {
+			const validatedEvent = this._eventValidator(nextEvent);
+			const validatedDate = this._dateValidator(nextEvent.date, nextEvent.time);
+			const events = this._events;
+			const index = events.findIndex((el) => el.id === id);
+			if (index !== -1) {
+				let callback;
+				if (!nextCallback) {
+					callback = events[index].callback;
+				} else {
+					callback = nextCallback;
+				}
+				const date = events[index].date + 'T' + events[index].time;
+				if (Date.parse(validatedDate) !== Date.parse(date)) {
+					clearTimeout(events[index].timerId);
+					const timerId = setTimeout(
+						callback,
+						new Date(validatedDate).getTime() - Date.now()
+					);
+					console.log(callback);
+					events[index] = {
+						...events[index],
+						...validatedEvent,
+						timerId,
+						callback,
+					};
+				}
+				events[index] = {
+					...events[index],
+					...validatedEvent,
+					callback,
+				};
+				return 'Updated successfully!';
+			} else {
+				throw new Error('Event not found!');
+			}
+		} catch (error) {
+			console.log(error.message);
+		}
+	}
+
+	/**
+	 * The method returns array with events objects in the specified date range
+	 * @method getEventsForPeriod
```

Comment:
`@method getEventsForPeriod` - redundant description, also in other places.

Reply:
Fixed